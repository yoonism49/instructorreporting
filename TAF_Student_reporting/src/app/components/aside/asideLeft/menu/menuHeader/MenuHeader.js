// @flow weak

import React            from 'react';
import PropTypes        from 'prop-types';
import {
  Motion,
  spring
}                       from 'react-motion';

const MenuHeader = ({
  title,
  backColor,
  isCollapsed,
  onClick
}) => (
  <ul
    className="list-inline sidebar-menu__group-container"
    onClick={onClick}>
    <li style={{width: '100%', backgroundColor: backColor}}>
      <a
        className="btn"
        style={{width: '100%', paddinLeft: '0px !important'}}>
        <span className="pull-left sidebar-menu__group-title">
          {title}
        </span>
        <Motion
          style={{
            deg: isCollapsed ? spring(0) : spring(180)
          }} >
          {
            ({ deg }) => {
              return (
                <i
                  className="fa fa-angle-up fa-1x pull-right"
                  style={{
                    WebkitTransform:  `rotate(${deg}deg)`,
                    transform:        `rotate(${deg}deg)`
                  }}
                  aria-hidden="true">
                </i>
              );
            }
          }
        </Motion>
      </a>
    </li>
  </ul>
);

MenuHeader.propTypes = {
  backColor:    PropTypes.string.isRequired,
  title:        PropTypes.string.isRequired,
  isCollapsed:  PropTypes.bool.isRequired,
  onClick:      PropTypes.func.isRequired
};

export default MenuHeader;
