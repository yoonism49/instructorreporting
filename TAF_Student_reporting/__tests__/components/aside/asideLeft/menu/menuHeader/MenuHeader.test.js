'use strict';

import React          from 'react';
import MenuHeader     from '../../../../../../src/app/components/aside/asideLeft/menu/menuHeader/MenuHeader';
import renderer       from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme

describe('AnimatedView component', () => {
  it('renders as expected', () => {
    const onClick = jest.fn();
    const component = renderer.create(
      <div>
        <MenuHeader
          title="menuHeaderTitle"
          backColor="#F1F2F3"
          isCollapsed
          onClick={onClick}
        />
      </div>
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
