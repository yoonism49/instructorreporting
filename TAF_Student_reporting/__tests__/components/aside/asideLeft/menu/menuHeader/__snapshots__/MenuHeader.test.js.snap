// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedView component renders as expected 1`] = `
<div>
  <ul
    className="list-inline sidebar-menu__group-container"
    onClick={[Function]}
  >
    <li
      style={
        Object {
          "backgroundColor": "#F1F2F3",
          "width": "100%",
        }
      }
    >
      <a
        className="btn"
        style={
          Object {
            "paddinLeft": "0px !important",
            "width": "100%",
          }
        }
      >
        <span
          className="pull-left sidebar-menu__group-title"
        >
          menuHeaderTitle
        </span>
        <i
          aria-hidden="true"
          className="fa fa-angle-up fa-1x pull-right"
          style={
            Object {
              "WebkitTransform": "rotate(0deg)",
              "transform": "rotate(0deg)",
            }
          }
        />
      </a>
    </li>
  </ul>
</div>
`;
